packages/flutter_tools/lib/src/platform_plugins.dart

```diff
@@ -230,40 +230,45 @@ class MacOSPlugin extends PluginPlatform {
 
 /// Contains the parameters to template a Windows plugin.
 ///
-/// The required fields include: [name] of the plugin, and [pluginClass] that will
-/// be the entry point to the plugin's native code.
+/// The [name] of the plugin is required. Either [dartPluginClass] or [pluginClass] are required.
+/// [pluginClass] will be the entry point to the plugin's native code.
 class WindowsPlugin extends PluginPlatform {
   const WindowsPlugin({
     @required this.name,
-    @required this.pluginClass,
+    this.pluginClass,
+    this.dartPluginClass,
   });
```

Comment:
Add an assert on one or the other being set since we don't have the annotations to catch issues any more.